### ISSUE TITLE
_childwindow_fix icompatible with gallium nine since WINE v7.3

### ISSUE THREAD

**Issue by leipero** [Frogging-Family/wine-tkg-git#716]
Since WINE v7.3, _childwindow_fix is not compatible with gallium nine. I know other vk related stuff is not compatible (fshack for example), the question is, is it still needed?

To avoid opening another issue, why is FUTEX2 moved to legacy? My understanding is that _fsync_futex_waitv="true" still requires FUTEX2, since implementation of waitv is for FUTEX2?

Side note: _use_josh_flat_theme="true" seems unnecessary since WINE v7.4.

**Comment by Tk-Glitch** [comment]
Vk child window support is needed for some apps/games, either natively using vulkan or when using dxvk. Gallium nine is a corner case that doesn't fall into either category (and I admit I haven't felt the need to use it for years). The newer version of the patchset is based on the fshack implementation, which might be why gnine isn't happy about it. If your GPU doesn't support Vulkan it's totally useless to you.

Futex2 was refused upstream and is now dead (and forcefully disabled when using futex_waitv). Futex_waitv supersedes futex2 and the only common dependency is fsync support (fsync itself depending on esync).

Josh's flat theme is still preferred by most users vs built-in theme so it's all subjective and not really important imho :P

**Comment by leipero** [comment]
Got it, then, I can close it, just wanted to point out that it became an issue recently. My GPU supports VK ofc., however, there are cases where gallium nine is prefered over DXVK (some really complex TMX tracks for example, where translation layer bottlenecks it regardless of how fast/slow CPU is), and since I didn't test since v7.2 (or earlier) I figured it out only yesterday. Now when you explained it, it makes sense, gallium nine isn't happy with fshack at all, and same type of crash happens with recent childwindow_fix.

Thanks, I got confused with futex2 to be frank, especially when you look at poronix articles and names of them (example: FUTEX2 futex_waitv Wired Up For Other Architectures With Linux 5.16-rc3), so I figured it's part of FUTEX2.

True, it is subjective, but it seems that new theme/style is absolutely fine and very similar to Josh's flat theme, I'm hard pressed to see a difference to be honest :).
